packages/services/cdn-worker/src/aws.ts

```diff
@@ -196,7 +196,26 @@ export class AwsClient {
             result: { type: 'success', response },
           });
 
-          return response;
+          if (!response.body) {
+            return response;
+          }
+
+          // Read the body first, to make sure it won't be aborted by the signal (timeout)
+          // If we get a timeout error during the body reading, we can't retry,
+          // as it's out of the control of that function.
+          // One example of it is when the res.text() is called with a delay due to application logic,
+          // and the timeout is reached at that point. The read will be aborted.
+          //
+          // For this reason, I prefer to read the body first, make it part of the retry logic,
+          // and only then return the response to the consumer,
+          // even at the cost of higher memory footprint (like it matters...).
+          const bodyText = await response.text();
```

Comment:
Awaiting makes sense to handle the errors, but should a `TypeError` be handled and not retried?
https://developer.mozilla.org/en-US/docs/Web/API/Response/text

And if you want to return a copy of the response, could you use `response.clone()`?

Reply:
Cloning a response has the same behavior, it's still affected by signals, so when a body is read the signal timeout is still part of it

Reply:
Sometimes the body is corrupted, retrying I think makes sense in this situation